nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py

```diff
@@ -66,6 +84,10 @@ def __init__(
         if isinstance(datasets[0], dict):
             self.load_data(datasets)
 
+        # Dataset is a list of json dicts as strings
```

Comment:
This part changes the original TextGeneration API.  In the API, it states
```                1. List of strings. Each element of the list provides input prompt. The model will apply tokenizer on it.
                    E.g [‘sentence’, ‘sentence2’ … ]
                                    2. Tuple of Pytorch Tensors (context_tokens, context_lengths). The `context_tokens` has shape (batch_size, seq_length),  it's the batched sequences of tokens used as a prompst for the generation or as model inputs to the encoder. 
                   The generative model will skip the tokenization and padding step.  The `context_lengths` has shape (batch_size,), it indicates the length of the context tokens for each of the input sequences.
                    E.g. ( torch.tensor([[23,5234,23,35,…], [223,323,23,23232,232,...] …]), torch.tensor([20, 30, …]))
                3. List of python dict objects. Used for prompt/p-tuning inputs where a set of key-value pairs are converted into input token embeddings for the model.
                    E.g. [{"prompt-tag": "sentiment", "sentence": "this is a good movie"},
                          {"prompt-tag": "qa", "context": "some context text", "question": "a simple question"} ... ]
                          where 'prompt-tag' is used to identify the type of NLP task to solve.
```
In your changes, the list of string can be either list of serialized python object or list of paths.  I would suggested for consistence,  use list of python dict for both of these cases. You can do 
 [ {'datapath': 'path str'}, {'datapath', 'path str'} ]
 or 
[ {'json_str': 'json str'}, {'json_str': 'json str' }]
which makes the type clear and avoid some hack like 'taskname' in str checks, which is error prone.

Reply:
I actually realized I don't need to support type (1) or (2) in the code anymore. So datasets is just (3) paths to .jsonl or .json files.